src/shapes/itext.class.js

```diff
@@ -510,24 +502,38 @@
      * @private
      * @param {CanvasRenderingContext2D} ctx
      */
-    _renderDragStartSelection: function (ctx) {
+    _renderDragSourceEffect: function (ctx) {
       if (this.__dragStartSelection) {
         this._renderSelection(
+          ctx,
           this.__dragStartSelection,
-          this._getCursorBoundaries(this.__dragStartSelection.selectionStart, true),
-          ctx
+          this._getCursorBoundaries(this.__dragStartSelection.selectionStart, true)
         );
       }
     },
 
+    renderDropTargetEffect: function(e, dontClear) {
```

Comment:
i tried it but didn't work, there was a case in which i was killing the source effect anyway.

What we can do is trying to transform ctx here, and pass it along and make that the effects receive a ctx.

move clearContextTop from text to object

clearContextTop for source, for target,
then draw them in sequence.

This will mean that we change the signature of the functions source and target to accept e and ctx.
is that ok?

Reply:
sounds better IMO